protos/flyteidl/plugins/sagemaker/trainingjob.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package flyteidl.plugins.sagemaker;
+
+option go_package = "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins";
+
+
+message AlgorithmSpecification {
+    string training_image = 1;  // TODO: will this change output? should it be an input instead of a part of the task spec?
+    string training_input_mode = 2;
+    string algorithm_name = 3;   // TODO: will this change output? should it be an input instead of a part of the task spec?
+
+    message MetricDefinition {
+        string name = 1;
+        string regex = 2;
+    }
+    repeated MetricDefinition metric_definitions = 4;
+}
+
+message ResourceConfig {
```

Comment:
this is interesting. I want to see how can we remove this from the users perspective too

Reply:
We might have to maintain a lookup table that maps from `(cpu, memory, gpu)x(request, limit)` to `instance type`. But `instance count` seems to be pretty implicit.